packages/contracts-bedrock/src/L2/L1Block.sol

```diff
@@ -73,4 +85,31 @@ contract L1Block is ISemver {
         l1FeeOverhead = _l1FeeOverhead;
         l1FeeScalar = _l1FeeScalar;
     }
+
+    /// @notice Updates the L1 block values for a post-blob activated chain.
+    /// Params are passed in as part of msg.data in order to compress the calldata.
+    /// Params should be passed in in the following order:
+    ///   1. _basefeeScalar      L1 basefee scalar
+    ///   2. _blobBasefeeScalar  L1 blob basefee scalar
+    ///   3. _sequenceNumber     Number of L2 blocks since epoch start.
+    ///   4. _number             L1 blocknumber.
+    ///   5. _timestamp          L1 timestamp.
+    ///   6. _basefee            L1 basefee.
+    ///   7. _blobBasefee        L1 blobBasefee.
+    ///   8. _hash               L1 blockhash.
+    ///   9. _batcherHash        Versioned hash to authenticate batcher by.
+    function setL1BlockValuesEcotone() external {
+        require(msg.sender == DEPOSITOR_ACCOUNT, "L1Block: only the depositor account can set L1 block values");
+        uint256 data;
+        assembly {
+            data := calldataload(4)
+            sstore(basefeeScalar.slot, shr(192, data)) // basefeeScalar & blobBaseFeeScalar (uint32,uint32)
+            sstore(sequenceNumber.slot, shr(128, data)) // uint64
```

Comment:
nit: a bit easier to read IMO/more consistent with how we do the rest of the sstores
```suggestion
            sstore(basefeeScalar.slot, shr(192, calldataload(4))) // basefeeScalar & blobBaseFeeScalar (uint32,uint32)
            sstore(sequenceNumber.slot, shr(128, calldataload(4))) // uint64
```

Reply:
I had pulled that out to avoid the gas cost of calling calldataload(4) twice, though I guess it's possible the compiler takes care of that?

Reply:
The compiler isn't smart enough to do things like that unfortunately

Reply:
@roberto-bayardo @tynes I was assuming what I suggested was pretty much equivalent in gas cost if not cheaper (since it's loading calldata), is that not the case?